### ISSUE TITLE
Something of with releases

### ISSUE THREAD

**Issue by yarikoptic** [datalad/git-annex#81]
GitHub says we have 430 releases.

It seems were have draft and released releases for the same latest tag
![Screenshot_20210902-220714](https://user-images.githubusercontent.com/39889/131939855-ec561b50-7f12-4502-bea5-1c36498ee54f.png)
And if I go to next page, it just keeps repeating - big in GitHub or something else - do we may be keep adding releases somehow @jwodder ?

**Comment by jwodder** [comment]
It appears that the `gh release` command on macOS (and only macOS?) keeps running into some race condition where it's apparently creating a draft version and then trying to publish it, but another build has already published a release for that tag.

I also note that the GitHub Actions for the three git-annex releases for which this occurred ([Aug 3](https://github.com/datalad/git-annex/runs/3233024547?check_suite_focus=true), [Jul 14](https://github.com/datalad/git-annex/runs/3068848775?check_suite_focus=true), [Jun 30](https://github.com/datalad/git-annex/runs/2954430401?check_suite_focus=true)) were manually triggered by you.  Were you rerunning the jobs over & over?  That would explain why there are so many draft releases.

**Comment by yarikoptic** [comment]
Those published releases - I guess I just edited some of the available draft releases or tags (don't remember now) to make them published via GitHub web ui. I didn't mess with jobs AFAIK

**Comment by jwodder** [comment]
@yarikoptic It does appear that the multiple drafts shown in the web interface are a bug on GitHub's end; according to the REST API, there are only ten releases in total, including one draft for each of the three bugged releases.

**Comment by yarikoptic** [comment]
Cool! Could you please report to GitHub?

**Comment by jwodder** [comment]
@yarikoptic On closer inspection, I believe that the 403 number is due to GitHub also counting tags that don't have releases; compare https://github.com/borntyping/python-colorlog, which has 57 tags and 57 "releases", yet only 15 of those releases are proper GitHub releases.  The draft releases are not counted in the 403 number, and indeed (based on their URLs), there are only three draft releases anyway (and they're not visible to people not affiliated with the repository).  The only remaining question is whether repeating the draft versions on every page of releases is intentional on GitHub's part or not.

I did, however, report a bug in `gh`: https://github.com/cli/cli/issues/4270

**Comment by yarikoptic** [comment]
damn -- indeed there are 403 (now 404 since new release is out) tags! I have not thought there were so many releases

```shell
(git)lena:~datalad/git-annex[master]git
$> git tag | nl | tail
   403	debian/6.20180112-1
   404	generate
```

thanks for the bug report!

re Drafts on each page -- could you also reach out to github support? I see how it could be useful to have them listed first, but listing on each page makes little to no sense to me (and with sufficient number of drafts might make it impossible to see any other tag/release but drafts I guess ;))

**Comment by jwodder** [comment]
@yarikoptic I've filed a bug report with GitHub.  Let's see what happens.

In the meantime, we may want to at least upload the DMGs from the draft releases to the corresponding published releases so that users can find them.

**Comment by yarikoptic** [comment]
yeah. Could you do that please or we could just wait for users to complain ;)

I guess tinuous would not pick them up automagically for the past releases, right? I will then manually download/add them (eventually)... I do not think @joeyh is relying on "our" osx builds, only  for windows. For OSX - referes to brew, and for conda - I never finished the https://github.com/conda-forge/git-annex-feedstock/pull/107 where I wanted to provide them for the "nodep" build... so - no rush, and I now have the most recent build if I need it "personally" ;)

**Comment by jwodder** [comment]
@yarikoptic Reuploaded.  tinuous won't pick up the new assets.

**Comment by yarikoptic** [comment]
> @yarikoptic I've filed a bug report with GitHub. Let's see what happens.

anything happened @jwodder?

**Comment by jwodder** [comment]
@yarikoptic Aside from a form response telling me the ticket was routed to the Product team, nothing.

**Comment by jwodder** [comment]
@yarikoptic GitHub never got back to me, but it looks like they took our advice anyway, since Drafts are now only listed on the first page of Releases.

**Comment by jwodder** [comment]
@yarikoptic Should we delete the draft releases?  They don't really serve a purpose.
